fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1623,6 +1644,96 @@ private static void parseInclusionParameter(Class<?> resourceType, FHIRSearchCon
         }
     }
 
+    /**
+     * Retrieves the search include restrictions.
+     *
+     * @param resourceType
+     *            the resource type
+     * @return list of allowed search _include values, or null if no restrictions
+     * @throws Exception
+     *             an exception
+     */
+    private static List<String> getSearchIncludeRestrictions(String resourceType) throws Exception {
+
+        // Retrieve the "resources" config property group.
+        PropertyGroup rsrcsGroup = FHIRConfigHelper.getPropertyGroup(FHIRConfiguration.PROPERTY_RESOURCES);
```

Comment:
We probably don't want to recompute this every time, is there a look aside map or structure we could use? 
Or should we cache this for the whole system on startup?

Reply:
I based this on SearchUtil.getFilterRules, which calls FHIRConfigHelper.getPropertyGroup each time. Since the config file contents are cached, I thought it would be ok to follow that as an example.

Reply:
haha, I actually had this same thought while editing SearchUtil.getFilterRules.   I was deciding how much trouble it would be to try caching this and only refresh it when the config file changes.  I ended up deciding to leave that optimatization for a future change, although in this case its new code so may be worth figuring out the caching thing.  I wish we had an easy way to measure performance of the search component as I don't like to optimize prematurely.

Reply:
OK - let's take a backlog item on this.  Anyone up for creating an issue for it? 

More as an investigation task?

Reply:
I opened https://github.com/IBM/FHIR/issues/1625 for the potential improvement.

Reply:
Lee opened #1625